### PR TITLE
RPI-213   Calc Engine Changes : HOLIDAY PAY ACCRUED section

### PR DESCRIPTION
**Pull request by Suprita-NTTDATA**
As the Calculation engine I want to change the calculation for Holiday Pay Accrued if the Claimant is an Irregular hours worker so that the new legislation is considered and used for calculations.

### CODE REVIEW

**Review by nflann11 — APPROVED** [review]
Code is well structured and well written and has sufficient testing

**Review by nflann11 — APPROVED** [review]

**Comment by Suprita-NTTDATA** [comment]
Approved sucessfully.

**Comment by Suprita-NTTDATA** [comment]
Approved sucessfully.

File: Insolvency.CalculationsEngine.Redundancy.API/Startup.cs [hunk 1]
```diff
@@ -3,6 +3,8 @@
 using Insolvency.CalculationsEngine.Redundancy.API.Infrastructure;
 using Insolvency.CalculationsEngine.Redundancy.API.Infrastructure.Middlewares;
 using Insolvency.CalculationsEngine.Redundancy.API.Infrastructure.Middlewares.Validators;
+using Insolvency.CalculationsEngine.Redundancy.API.Infrastructure.Middlewares.Validators.IrregaulrWorkerHourHPA;
```
Comment (gleninsolvencyservice) [inline Insolvency.CalculationsEngine.Redundancy.API/Startup.cs:6]: Is this just a misspell? if so, can it be removed.

File: Insolvency.CalculationsEngine.Redundancy.BL/Services/Implementations/HolidayCalculationService.cs [hunk 2]
```diff
@@ -29,11 +30,29 @@ public async Task<HolidayCalculationResponseDTO> PerformHolidayCalculationAsync(
             if (data.Hpa != null)
                 return await PerformHolidayCalculationWithHPAAsync(data, options);
             else if (data.Htnp != null && data.Htnp.Any())
-                return await PerformHolidayCalculationWithoutHPAAsync(data, options);
+                return await PerformHolidayCalculationWithoutHPAAsync(data.Htnp, options);
```
Comment (gleninsolvencyservice) [inline Insolvency.CalculationsEngine.Redundancy.BL/Services/Implementations/HolidayCalculationService.cs:33]: What is the reasoning behind the change to use Htnp?
Is this a change to existing code?

Comment (Suprita-NTTDATA) [reply Insolvency.CalculationsEngine.Redundancy.BL/Services/Implementations/HolidayCalculationService.cs:33]: To reuse this piece of code in PerformIrregularHolidayCalculationAsync method for calculating.

File: Insolvency.CalculationsEngine.Redundancy.API/Startup.cs (hunk 1 above)
Comment (Suprita-NTTDATA) [reply Insolvency.CalculationsEngine.Redundancy.API/Startup.cs:6]: Yes . Good catch I will rename it and push it again. Thanks

**Pull request merged**